### PR TITLE
Use merr in proxy

### PR DESCRIPTION
**Pull request by smellthemoon**
Signed-off-by: lixinguo [xinguo.li@zilliz.com](mailto:xinguo.li@zilliz.com)
/kind improvement

### CODE REVIEW

**Comment by smellthemoon** [comment]
rerun ut

**Comment by smellthemoon** [comment]
rerun ut

File: internal/proxy/authentication_interceptor.go [hunk 1]
```diff
@@ -63,7 +64,7 @@ func AuthenticationInterceptor(ctx context.Context) (context.Context, error) {
 		if !validSourceID(ctx, md[strings.ToLower(util.HeaderSourceID)]) {
 			username, password := parseMD(md[strings.ToLower(util.HeaderAuthorize)])
 			if !passwordVerify(ctx, username, password, globalMetaCache) {
-				return nil, ErrUnauthenticated()
+				return nil, merr.WrapErrIoFailed("metadata", "auth check failure, please check username and password are correct")
```
Comment (yah01) [inline internal/proxy/authentication_interceptor.go:67]: It should be ErrParameterInvalid

File: internal/util/merr/errors.go [hunk 2]
```diff
@@ -96,7 +96,10 @@ var (
 	ErrParameterInvalid = newMilvusError("invalid parameter", 1100, false)
 
 	// Metrics related
-	ErrMetricNotFound = newMilvusError("MetricNotFound", 1200, false)
+	ErrMetricNotFound = newMilvusError("metric not found", 1200, false)
+
+	// Memory related
+	ErrMemoryNotSufficient = newMilvusError("memory not sufficient to load", 1300, false)
```
Comment (yah01) [inline internal/util/merr/errors.go:102]: Redundancy of ErrServiceMemoryLimitExceeded

**Review by yah01 — CHANGES_REQUESTED** [review]

File: internal/proxy/impl.go [hunk 3]
```diff
@@ -1452,16 +1453,8 @@ func (node *Proxy) GetLoadingProgress(ctx context.Context, request *milvuspb.Get
...
-			return &milvuspb.GetLoadingProgressResponse{
-				Status: InSufficientMemoryStatus(request.GetCollectionName()),
-			}
-		}
 		return &milvuspb.GetLoadingProgressResponse{
-			Status: &commonpb.Status{
-				ErrorCode: commonpb.ErrorCode_UnexpectedError,
-				Reason:    err.Error(),
-			},
+			Status: merr.Status(err),
```
[the remark this answers is not in the document]
Comment (smellthemoon) [reply internal/proxy/impl.go:1462]: ok, will keep the error not change until qn has processed.

**Review by jiaoew1991 — APPROVED** [review]
/lgtm

**Pull request merged**